File: streampipes-sdk/src/main/java/org/apache/streampipes/sdk/utils/Datatypes.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package org.apache.streampipes.sdk.utils;

import org.apache.streampipes.vocabulary.SO;
import org.apache.streampipes.vocabulary.XSD;

import java.net.URI;
import java.util.Arrays;

public enum Datatypes {

  Integer(XSD.INTEGER),
  Long(XSD.LONG),
  Float(XSD.FLOAT),
  Boolean(XSD.BOOLEAN),
  String(XSD.STRING),
  Double(XSD.DOUBLE),
  Number(URI.create(SO.NUMBER)),
  Sequence(XSD.SEQUENCE);

  private URI uri;

  Datatypes(URI uri) {
    this.uri = uri;
  }

  public static Datatypes fromDatatypeString(String datatype) {
    return Arrays.stream(Datatypes.values())
        .filter(d -> d.uri.toString().equals(datatype))
        .findFirst()
        .orElseThrow(() -> new RuntimeException("Could not find datatype with URI " + datatype));
  }

  public String toString() {
    return uri.toString();
  }
}

File: ui/src/app/dashboard/sdk/model/datatypes.ts
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

import { Vocabulary } from './vocabulary';

export class Datatypes {
    static readonly Integer = new Datatypes(Vocabulary.XSD, 'integer');
    static readonly Long = new Datatypes(Vocabulary.XSD, 'long');
    static readonly Float = new Datatypes(Vocabulary.XSD, 'float');
    static readonly Boolean = new Datatypes(Vocabulary.XSD, 'boolean');
    static readonly String = new Datatypes(Vocabulary.XSD, 'string');
    static readonly Double = new Datatypes(Vocabulary.XSD, 'double');
    static readonly Number = new Datatypes(Vocabulary.SO, 'Number');

    private constructor(
        private readonly vocabulary: string,
        private readonly entity: string,
    ) {}

    toUri() {
        return this.vocabulary + this.entity;
    }
}
